internal/services/containers/kubernetes_flux_configuration_resource.go

```diff
@@ -797,13 +796,13 @@ func expandAzureBlobDefinitionModel(inputList []AzureBlobDefinitionModel) (*flux
 	}
 
 	if input.ContainerID != "" {
-		id, err := parse.StorageContainerDataPlaneID(input.ContainerID)
+		id, err := containers.ParseContainerID(input.ContainerID, storageDomainSuffix)
 		if err != nil {
 			return nil, err
 		}
 
-		output.ContainerName = &id.Name
-		output.Url = pointer.To(strings.TrimSuffix(input.ContainerID, "/"+id.Name))
+		output.ContainerName = &id.ContainerName
+		output.Url = pointer.To(strings.TrimSuffix(input.ContainerID, "/"+id.ContainerName))
```

Comment:
perhaps pedantic but couldn't we make this:

```suggestion
		output.Url = pointer.To(fmt.Sprintf("https://%s", id.Domain))
```

(or add a function to return this value?)

That'd allow the usage of this Resource ID to be better highlighted, to help us determine the impact of a change?